timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -617,10 +628,39 @@ export class TimeGraphChart extends TimeGraphChartLayer {
         el.displayObject.on('click', ((e: PIXI.InteractionEvent) => {
             if (el instanceof TimeGraphStateComponent && !this.mousePanning && !this.mouseZooming) {
                 this.selectState(el.model);
+                // Time range selection using left-click should not trigger a double click action
+                if (!(this.unitController.selectionRange && (this.unitController.selectionRange.start !== this.unitController.selectionRange.end))) {
+                    if (el.clickStatus){
+                        // perform double click action
+                        // changing event type from 'click' to 'doubleClick'
+                        e.type = 'doubleClick';
+                        // reset counter
+                        el.clickStatus = false;
+                    }else{
+                        // perform single click action
+                        // set counter & keep track of recently clicked element
+                        if (this._recentlyClickedElement) {
+                            this._recentlyClickedElement.clickStatus = false;
+                        }
+                        el.clickStatus = true;
+                        this._recentlyClickedElement = el;
+    
+                        setTimeout(function() {
+                            // after action performed, reset counter 
+                            if (el.clickStatus){
+                                el.clickStatus = false; 
+                            }
+                        }, this._dbClickTime);          
+                    }
+                }
             }
-            if (this.mouseInteractions && this.mouseInteractions.click) {
+
+            if (e.type === 'click' && this.mouseInteractions && this.mouseInteractions.click) {
```

Comment:
Looking at this for example: https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event

It seems like on the doubleClick we should also see a click event just before. e.g. click(1), click(2), doubleClick.

When I triple-, quadruple-click and more, it seems that doubleClick is sporadically sent, and it depends on the timing. There should be only one doubleClick, e.g. click(1), click(2), doubleClick, click(3), click(4), click(5), ...

I would propose that this class just keeps track locally of the count. The count keeps increasing without limit as long as we keep clicking on the same coordinate. We can use a debouncer to reset the count when no click occurs for a certain period. Then the doubleClick event would be fired only when count == 2.

I would also propose to improve the click callback to include the count parameter (unless we can somehow include the count in the event, but not sure if good idea or even possible to modify it since we didn't create it).

Reply:
Thanks for the feedback Patrick! The implementation looks much cleaner now. Let me know if you need any additional changes